### ISSUE TITLE
[Analytic Privileges] Synchronizer

### ISSUE THREAD

**Issue by dpanayotov** [SAP/xsk#902]
# Target
When publishing a project, the analytic privileges need to be created in the HDI container.
As the analytic privilege file needs to be listed in the `hdi` file, the processing of the analytic privilege should happen when processing the hdi container

**Comment by aihanrashidov** [comment]
The analytic privileges seem to be processed successfully within the hdi containers.
Steps to reproduce:
1. Create a simple hdi project or use one of the samples - [hdi-simple](https://github.com/SAP/xsk/tree/main/samples/hdi-simple) / [hdi-cube](https://github.com/SAP/xsk/tree/main/samples/hdi-cube) / [hdi-ext](https://github.com/SAP/xsk/tree/main/samples/hdi-ext).
2. Add the required analytic privilege plugin to the `.hdiconfig` file.
<img width="499" alt="Screenshot 2022-01-04 at 11 59 34" src="https://user-images.githubusercontent.com/36267628/148042721-a5931244-c622-43d9-b16b-198e2825b9e7.png">

3. Create an analytic privilege file - `test.hdbanalyticprivilege` and specify the secured model (the calculation view in our case).
<img width="1300" alt="Screenshot 2022-01-04 at 11 58 59" src="https://user-images.githubusercontent.com/36267628/148042865-1e7d738e-7a9a-418e-bcf4-7a049df6b56d.png">

4. Add the analytic privilege file to the deploy paths.
<img width="464" alt="Screenshot 2022-01-04 at 12 07 17" src="https://user-images.githubusercontent.com/36267628/148042969-86b23a6d-58eb-43c0-869b-aea03b09cd4f.png">

5. Set the `.hdbcalculationview` to be secured by analytic privilege.
<img width="881" alt="Screenshot 2022-01-04 at 12 00 03" src="https://user-images.githubusercontent.com/36267628/148043326-ef2db74a-8751-4924-90a8-556765a53a05.png">

6. Publish the project.

7. if there are no errors in log check if the privilege is created in the hana cockpit.
<img width="1078" alt="Screenshot 2022-01-04 at 12 12 34" src="https://user-images.githubusercontent.com/36267628/148043627-8eb0c3d9-b366-4d21-8c57-abc6a617b0e8.png">

**Comment by ThuF** [comment]
@aihanrashidov great!!! Thank you for the update! Now the next step is to automate the manual steps to populate the `.hdiconfig` and `.hdi`files.

**Comment by Fluctuationqt** [comment]
In BAS generated projects have ```.hdiconfig``` syntax like this, which does not include specific versions of the plugins. Only a 
```
// .hdiconfig
{
	"minimum_feature_version": "1006",
	"file_suffixes": {
		"hdbapplicationtime": {
			"plugin_name": "com.sap.hana.di.applicationtime"
		},
		"hdbcalculationview": {
			"plugin_name": "com.sap.hana.di.calculationview"
		}
}
```

**Comment by vmutafov** [comment]
@aihanrashidov for the changes in the Calculation view, a similar XSLT could be used:
```JS
transformColumnObject(calculationViewXmlBytes) {
        const columnObjectToResourceUriXslt = `<?xml version="1.0" encoding="ISO-8859-1"?>
            <xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">

                <xsl:template match="node()|@*">
                    <xsl:copy>
                        <xsl:apply-templates select="node()|@*"/>
                    </xsl:copy>
                </xsl:template>

                <xsl:template match="DataSource[@type='DATA_BASE_TABLE']/columnObject[@columnObjectName]">
                    <xsl:element name="resourceUri">
                        <xsl:value-of select="@columnObjectName"/>
                    </xsl:element>
                </xsl:template>
            </xsl:stylesheet>
        `;

        const factory = TransformerFactory.newInstance();
        const source = new StreamSource(new StringReader(columnObjectToResourceUriXslt));
        const transformer = factory.newTransformer(source);

        const text = new StreamSource(new ByteArrayInputStream(calculationViewXmlBytes));

        const bout = new ByteArrayOutputStream();
        transformer.transform(text, new StreamResult(bout));
        return bout.toByteArray();
    }
```

This is the JS code for transforming the column objects when processing the calculation views in the migration backend. This could easily be done in Java as well as those APIs used for XSLT transformations come from the Java standard library.
These docs are helpful for the XSLTs: 
- https://docs.oracle.com/javase/tutorial/jaxp/xslt/transformingXML.html
- https://mkyong.com/java/java-dom-parser-xml-and-xslt-examples/

And this one for testing some XSLTs quickly: https://www.freeformatter.com/xsl-transformer.html

**Comment by vmutafov** [comment]
`.hdbanalyticprivilege` and `.analyticprivilege` files are added in the deployables with this PR: https://github.com/SAP/xsk/pull/997

**Comment by ThuF** [comment]
Closed with #997
